python/salpytools/salpylib.py

```diff
@@ -122,40 +123,64 @@ def subscribe_logEvent(self, eventname):
         self.mgr[eventname].salEvent("{}_logevent_{}".format(self.subsystem_tag, eventname))
         self.logEvent[eventname] = getattr(self.mgr[eventname],'logEvent_{}'.format(eventname))
         self.myData[eventname] = getattr(self.SALPY_lib,'{}_logevent_{}C'.format(self.subsystem_tag, eventname))()
-            
+
         self.myData_keys[eventname] = [a[0] for a in inspect.getmembers(self.myData[eventname]) if not(a[0].startswith('__') and a[0].endswith('__'))]
         self.log.info('Initializing: {}_logevent_{}'.format(self.subsystem_tag, eventname))
-        
+
     def get_current_state(self):
         """
         Function to get the current state
 
         :return:
         """
-        return self.current_state
+        return self.model.state
 
     def validate_transition(self, new_state):
 
         return validate_transition(self.current_state, new_state)
 
-    def make_transition(self, new_state):
+    def execute_command(self, command):
+        """This method delegates commands recieved by a DDSController to the
+        state. 
 
-        if not self.validate_transition(new_state):
-            raise Exception("Cannot perform transition: {} --> {}".format(self.current_state, new_state))
-        else:
-            self.states[self.current_state].sleep()
-            self.current_state = new_state
-            self.states[new_state].wake()
+        The model is passed so that the State object may call methods on
+        it. Also the state is stored on the model only as a string
+        representation. The actual state object is stored on this context
+        object as self.states. 
+
+        Attributes:
+            command: A string representation of the command recieved by a
+            DDSController object.
+        """
+
+        current_state = self.states[self.model.state]
+
+        if command == "ENTERCONTROL":
+            current_state.enter_control(self.model)
```

Comment:
Wouldn't it be better to add the model to the state instead of passing it along on the call to the method? If it is something that is really needed (and I agree it is) I would rather have it as part of the class already.

Reply:
Yes, I considered this but I was unsure if doing so might be a gotcha later down during design so I had not done so.

I agree that it makes more sense to put it part of the class, since even you saw this I have added the model to the class. 

If it somehow fundamentally changes architecture in a way that negatively impacts the software I will leave that to Paul and whoever else when we do a design review.